testsuite/tests/weak-ephe-final/weaklifetime_par.ml

```diff
@@ -76,28 +70,32 @@ let check_and_change data i j =
 
 let dummy = ref [||];;
 
+
 let run index () =
   let domain_data = Array.init 100 (fun i ->
-    let n = 1 + random_int 100 in
+    let n = 1 + Random.int 100 in
     {
       objs = Array.make n (Absent 0);
       wp = Weak.create n;
     }
   ) in
-  while gccount () < 5 do
-    dummy := Array.make (random_int 300) 0;
-    let i = (random_int (size/num_domains)) + index * size/num_domains in
-    let j = random_int (Array.length data.(i).objs) in
+  let gc_start = gccount () in
+  while gccount () - gc_start < 5 do
+    dummy := Array.make (Random.int 300) 0;
+    let per_domain_size = size / num_domains in
+    assert (index < num_domains);
+    let i = index * per_domain_size + Random.int per_domain_size in
+    let j = Random.int (Array.length data.(i).objs) in
     check_and_change data i j;
-    let ix = random_int 100 in
-    let jx = random_int (Array.length domain_data.(ix).objs) in
+    let ix = Random.int 100 in
+    let jx = Random.int (Array.length domain_data.(ix).objs) in
     check_and_change domain_data ix jx
   done
 
 let _ =
-  for index = 0 to 4 do
-    let domains = Array.init (num_domains - 1) (fun i -> Domain.spawn(run ((i + index) mod 5))) in
-    run ((num_domains - 1 + index) mod 5) ();
+  for index = 0 to num_domains - 1 do
```

Comment:
I think with this we are going through the loop one less time than before. (0 to 4 vs 0 to 3 now).
Which may be fine, since before your fix the amount of work done was quite limited anyway.

Reply:
At the time where I made this change, I had the impression that it was necessary for indexing correctness.

Thinking about it now, I don't think it's the case: now that the modulo is fixed, we could use any value here, we are just rotating the mapping from domains to array segments. The particular choice of `num_domains - 1` has the cute property that each domain will run with each segment exactly once, but this is not necessary. 

 may rework the code slightly to make the numbr of rounds here a completely independent parameter. (This avoids quadratic costs when increasing the domain count.) Thanks!

Reply:
It's an independent parameter now.